zstash/globus_utils.py

```diff
@@ -0,0 +1,273 @@
+from __future__ import absolute_import, print_function
+
+import configparser
+import json
+import os
+import os.path
+import re
+import socket
+import sys
+from typing import Dict, List, Optional
+
+from globus_sdk import (
+    NativeAppAuthClient,
+    RefreshTokenAuthorizer,
+    TransferAPIError,
+    TransferClient,
+    TransferData,
+)
+from globus_sdk.response import GlobusHTTPResponse
+
+from .settings import logger
+
+# Global constants ############################################################
+HPSS_ENDPOINT_MAP: Dict[str, str] = {
+    "ALCF": "de463ec4-6d04-11e5-ba46-22000b92c6ec",
+    "NERSC": "9cd89cfd-6d04-11e5-ba46-22000b92c6ec",
+}
+
+# This is used if the `globus_endpoint_uuid` is not set in `~/.zstash.ini`
+REGEX_ENDPOINT_MAP: Dict[str, str] = {
+    r"theta.*\.alcf\.anl\.gov": "08925f04-569f-11e7-bef8-22000b9a448b",
+    r"blueslogin.*\.lcrc\.anl\.gov": "15288284-7006-4041-ba1a-6b52501e49f1",
+    r"chrlogin.*\.lcrc\.anl\.gov": "15288284-7006-4041-ba1a-6b52501e49f1",
+    r"b\d+\.lcrc\.anl\.gov": "15288284-7006-4041-ba1a-6b52501e49f1",
+    r"chr.*\.lcrc\.anl\.gov": "15288284-7006-4041-ba1a-6b52501e49f1",
+    r"compy.*\.pnl\.gov": "68fbd2fa-83d7-11e9-8e63-029d279f7e24",
+    r"perlmutter.*\.nersc\.gov": "6bdc7956-fc0f-4ad2-989c-7aa5ee643a79",
+}
+
+ZSTASH_CLIENT_ID: str = "6c1629cf-446c-49e7-af95-323c6412397f"
+
+# State files
+GLOBUS_CFG: str = os.path.expanduser("~/.globus-native-apps.cfg")
+INI_PATH: str = os.path.expanduser("~/.zstash.ini")
+TOKEN_FILE = os.path.expanduser("~/.zstash_globus_tokens.json")
+
+# Independent functions #######################################################
+# The functions here don't rely on the global variables defined in globus.py.
+
+
+# Primarily used by globus_activate ###########################################
+def check_state_files():
+    if os.path.exists(GLOBUS_CFG):
+        logger.warning(
+            f"Globus CFG {GLOBUS_CFG} exists. This may be left over from earlier versions of zstash, and may cause issues. Consider deleting."
+        )
+
```

Comment:
We no longer need to worry about `GLOBUS_CFG`, resolving `(2) Delete existing globus cfg file` from #339